src/irmin-pack/checks.ml

```diff
@@ -38,52 +37,71 @@ let path =
 
 module Make (Args : Make_args) = struct
   module Hash = Args.Hash
-  module Store = Args.Store
+  module Store_V1 = Args.Store_V1
+  module Store_V2 = Args.Store_V2
   module Index = Pack_index.Make (Hash)
 
+  let current_version = `V1
+
   (** Read basic metrics from an existing store. *)
   module Stat = struct
     type size = Bytes of int [@@deriving irmin]
-
-    type io = { size : size; offset : int64; generation : int64 }
+    type version = [ `V1 | `V2 ] [@@deriving irmin]
+
+    type io = {
+      size : size;
+      offset : int64;
+      generation : int64;
+      version : version;
+    }
     [@@deriving irmin]
 
     type files = { pack : io option; branch : io option; dict : io option }
     [@@deriving irmin]
 
     type t = { hash_size : size; files : files } [@@deriving irmin]
 
-    let with_io : type a. string -> (IO.t -> a) -> a option =
-     fun path f ->
+    let with_io : type a. I.version -> string -> (IO.t -> a) -> a option =
+     fun version path f ->
       match IO.exists path with
       | false -> None
       | true ->
           let io =
-            IO.v ~fresh:false ~readonly:true ~version:(Some current_version)
-              path
+            IO.v ~fresh:false ~readonly:true ~version:(Some version) path
           in
           Fun.protect ~finally:(fun () -> IO.close io) (fun () -> Some (f io))
 
-    let io path =
-      with_io path @@ fun io ->
+    let detect_version root =
```

Comment:
Would it make sense to expose `detect_version` (or something similar) publicly? I'm not sure how versions are managed, so maybe that isn't very useful.

Reply:
Yes, it's now used by the layered/check, so its exposed.